File: frontend/src/microservices/operations/attraction/api/RegisterAttractionManagerPlanner.ts
// 鏈枃浠跺畾涔?RegisterAttractionManagerPlanner锛岃礋璐?operations 妯″潡鐨勬敞鍐岀紪鎺掑拰鎺ュ彛鍏ュ彛銆?

import { executeJsonApiRequest } from '@/shared-kernel/api/ApiTransport'
import type { RegisterAttractionManagerPlannerRequest } from '@/microservices/operations/attraction/objects/RegisterAttractionManagerPlannerRequest'
import type { AttractionManagerSessionPlannerResponse } from '@/microservices/operations/attraction/objects/AttractionManagerSessionPlannerResponse'

export const registerAttractionManager = (payload: RegisterAttractionManagerPlannerRequest): Promise<AttractionManagerSessionPlannerResponse> =>
  executeJsonApiRequest<AttractionManagerSessionPlannerResponse>('/RegisterAttractionManagerPlanner', 'POST', payload)

File: backend/src/main/scala/microservices/operations/attraction/api/RegisterAttractionManagerPlanner.scala
// RegisterAttractionManagerPlanner 负责operations相关实现。

package com.typesafe.travel.operations.domain

import cats.effect.IO
import com.typesafe.travel.api.routes.ConnectionApiPlan
import com.typesafe.travel.auth.domain.hashPasswordForLoginEmail
import com.typesafe.travel.persistence.operations.AttractionManagerPlainSql
import com.typesafe.travel.shared.kernel.EmailAddress

import java.sql.Connection
import java.time.Instant
import java.util.UUID

object RegisterAttractionManagerPlanner extends ConnectionApiPlan[RegisterAttractionManagerPlannerRequest, AttractionManagerSessionPlannerResponse]:
  override val name: String = "RegisterAttractionManagerPlanner"
  override def plan(input: RegisterAttractionManagerPlannerRequest, connection: Connection): IO[AttractionManagerSessionPlannerResponse] =
    val now = Instant.now()
    val managerId = s"attraction-manager-${UUID.randomUUID().toString.take(12)}"
    for
      email <- IO.fromEither(EmailAddress.create(input.email))
      _ <- validateRegisterAttraction(input)
      passwordHash <- hashPasswordForLoginEmail(input.password, email)
      _ <- AttractionManagerPlainSql.insertAttractionManager(connection, managerId, input.email, input.displayName, now)
      _ <- AttractionManagerPlainSql.insertAttractionManagerCredential(connection, managerId, input.email, passwordHash, now)
    yield AttractionManagerSessionPlannerResponse(managerId, "Attraction", input.email.trim, input.displayName.trim, "Active", managerId, None, now.toString)


